kernel/vmbytegen.ml

```diff
@@ -831,9 +831,15 @@ and compile_instance env cenv u sz cont =
     | None -> compile_structured_constant cenv (Const_quality q) sz cont
     | Some idx -> pos_instance cenv sz :: Kfield 0 :: Kfield idx :: cont
     in
-    let comp_univ cenv l sz cont = match Univ.Level.var_index l with
-    | None -> compile_structured_constant cenv (Const_univ_level l) sz cont
-    | Some idx -> pos_instance cenv sz :: Kfield 1 :: Kfield idx :: cont
+    (* todo fix MS *)
+    let comp_univ cenv u sz cont = 
+      match Univ.Universe.level u with
+      | Some l ->
+        (match Univ.Level.var_index l with
+        | None -> compile_structured_constant cenv (Const_univ u) sz cont
+        | Some idx -> pos_instance cenv sz :: Kfield 1 :: Kfield idx :: cont)
+      | None ->
+        compile_structured_constant cenv (Const_univ u) sz cont
```

Comment:
This is incorrect, a universe not being a level doesn't mean it's constant

Reply:
Yup